pkg/controller/upload_controller.go

```diff
@@ -327,6 +327,26 @@ func (c *uploadController) processUpload(req *pluginv1api.Upload) error {
 		return errors.New(errMsg)
 	}
 
+	// If upload phase is UploadPhaseCleanupFailed, manually call DeleteLocalSnapshot to retry on deleting local snapshot
+	if req.Status.Phase == pluginv1api.UploadPhaseCleanupFailed {
+		if c.snapMgr.DeleteLocalSnapshot(peID) != nil {
```

Comment:
Could you split this in 2 lines? one where you get the error, and the next where you check if err is not nil

Reply:
As this err is only used inside the if block, so here merge these two lines into one in order to avoid propagating the err value to the following context outside the if block.

Reply:
We lose the actual error thrown by `c.snapMgr.DeleteLocalSnapshot`.  In the errMsg what does `errors.WithStack(err)` point to?, i am guessing it'll always be nil and not the actual err returned by `c.snapMgr.DeleteLocalSnapshot`

Reply:
Agree with this. Split to 2 lines to record the actual error returned by DeleteLocalSnapshot